test/py/test_broadcast_perf.py

```diff
@@ -0,0 +1,288 @@
+    })
+
+    global_workitems = max(2, tensorsize / 32)
+
+    # construct an "interesting" range for global_workitems
+    gl_range = []
+    xx = tensorsize / 16
+    for i in range(5):
+        vv = int(3 * i * xx)
+        ww = int(2**i * xx)
+        if vv > 63:
+            gl_range.append(vv)
+            gl_range.append(ww)
+    gl_range.sort()
+    print(tensorsize, '------------------>gl_range=', gl_range)
+
+    # I don't know the exact limitation on global_workitems, but if it's bigger than roughly 1G it
+    # will cause a program crash.
+    for global_workitems in gl_range:
+        if global_workitems > 1024 * 1024 * 1022:
+            continue
+
+        # result items to save in JSON form
+        dataset = []
+
+        # result items to plot immediately (not save)
+        plot_set = []
+
+        # Any value larger than 1024 is rejected by hip library
+        for local_workitems_per_CU in range(64, 1025, 64):
+            print('local work items=', local_workitems_per_CU)
+            if (local_workitems_per_CU > global_workitems):
+                continue
+
+            #############################################
+            #  Generate a temporary JSON file for the gpu-driver
+            #############################################
+
+            cp = {}  # compile_pointwise key
+
+            # lambda to be executed by the operator.  This lambda contains enough looping to make operator time detectable
+            cp["lambda"] = "[](auto x, auto y, auto z) {for (int i = 1; i < 200; i++){ z = sqrt(abs(z+y));} return x+y+z; }"
+
+            # inputs is a list of dict
+            inputs = []
+
+            # 4 inputs, of which 2 are broadcasted if stride is set
+            item = {"type": datatype, "lens": myshape, "strides": {}}
+            inputs.append(item)
+
+            item = {"type": datatype, "lens": myshape, "strides": stride}
+            inputs.append(item)
+
+            item = {"type": datatype, "lens": myshape, "strides": stride}
+            inputs.append(item)
+
+            item = {"type": datatype, "lens": myshape, "strides": {}}
+            inputs.append(item)
+
+            cp["inputs"] = inputs
+
+            cp["global"] = global_workitems
+            cp["local"] = local_workitems_per_CU
+            cp["iterations"] = iterations
+
+            # top level JSON node
+            top_node = {"compile_pointwise": cp}
+
+            # Format as JSON and write to file
+            with open(os.path.join(save_python_dir, 'temp_file.json'),
+                      'w') as f:
+                f.write(json.dumps(top_node, indent=4))
+            # f.close() not necessary
+
+            ##################################################
+            #   run gpu-driver
+            ##################################################
+
+            # print('run the gpu-driver...')
+            stream = os.popen(os.path.join(base_dir,'build/bin/gpu-driver  ') +\
```

Comment:
We can't assume the build directory is located under build/. We should probably take the gpu-driver as an input parameter.